src/System.Text.Primitives/System/Text/InvariantParser_ulong.cs

```diff
@@ -30,20 +30,23 @@ public static bool TryParse(Utf8String utf8Text, out ulong value, out int bytesC
                         return true;
                     }
                 }
-                try
+                else if (value > UInt64.MaxValue / 10) // overflow
                 {
-                    ulong candidate = checked(value * 10 + nextByte - '0');
```

Comment:
that's a fabulous idea and we should get minor perf bonuses from doing less checks. Doing this now